### DISCUSSION TITLE
How can I secure MQTT connection ?

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [Koenkk/zigbee2mqtt#17836] [General]
Hi,
I just installed MQTT + Z2M in my docker compose stack.
My problem is that most of my containers are using docker bridge mode, so that there is no exposed ports.
Home Assistant needs to be in host mode to work correctly (for DLNA). It can't join bridge network.
To make MQTT accessible from HA, I need to expose MQTT port. I don't like doing that, especially without a password.
I want to set a password to MQTT, but I can't find how. I found no explaination for this in the docs. I found how to specify which login to use in Z2M : https://www.zigbee2mqtt.io/guide/configuration/mqtt.html#server-connection but not how to set this password in the first place.

Could someone help me ?
Thanks in advance, have a good day.

**Comment by Nardol** [comment]
Maybe this tutorial could help you: https://cedalo.com/blog/mosquitto-docker-configuration-ultimate-guide/
